satpy/writers/cf_writer.py

```diff
@@ -396,6 +350,106 @@ def make_alt_coords_unique(datas, pretty=False):
+    automatically.
+
+    """
+    for da_name, data in datas.items():
+        declared_coordinates = data.attrs.get('coordinates', [])
+        if isinstance(declared_coordinates, str):
+            declared_coordinates = declared_coordinates.split(' ')
+        for coord in declared_coordinates:
+            if coord not in data.coords:
+                try:
+                    dimensions_not_in_data = list(set(datas[coord].dims) - set(data.dims))
+                    data[coord] = datas[coord].squeeze(dimensions_not_in_data, drop=True)
+                except KeyError:
+                    warnings.warn(
+                        'Coordinate "{}" referenced by dataarray {} does not '
+                        'exist, dropping reference.'.format(coord, da_name),
+                        stacklevel=2
+                    )
+                    continue
+
+        # Drop 'coordinates' attribute in any case to avoid conflicts in xr.Dataset.to_netcdf()
+        data.attrs.pop('coordinates', None)
+
+
+# ###--------------------------------------------------------------------------.
+# ### CF-Time
+def make_time_bounds(start_times, end_times):
+    """Create time bounds for the current *dataarray*."""
+    start_time = min(start_time for start_time in start_times
+                     if start_time is not None)
+    end_time = min(end_time for end_time in end_times
+                   if end_time is not None)
+    data = xr.DataArray([[np.datetime64(start_time), np.datetime64(end_time)]],
+                        dims=['time', 'bnds_1d'])
+    return data
+
+
+def add_time_cf_attrs(ds):
+    """Add time CF-compliant attributes if Dataset has time coordinate."""
+    if 'time' in ds:
+        # Retrieve list of start_time and end_time
+        start_times = []
+        end_times = []
+        for _var_name, data_array in ds.items():
+            start_times.append(data_array.attrs.get("start_time", None))
+            end_times.append(data_array.attrs.get("end_time", None))
+        # Add time bounds dimension and bounds attribute
+        ds['time_bnds'] = make_time_bounds(start_times, end_times)
+        ds['time'].attrs['bounds'] = "time_bnds"
+        ds['time'].attrs['standard_name'] = "time"
+    # else:
+    #     grp_str = ' of group {}'.format(group_name) if group_name is not None else ''
+    #     logger.warning('No time dimension in datasets{}, skipping time bounds creation.'.format(grp_str))
+    return ds
+
+
+def _encode_time(dataarray, epoch):
+    """Add encodings to the 'time' coordinate."""
+    if 'time' in dataarray.coords:
+        dataarray['time'].encoding['units'] = epoch
+        dataarray['time'].attrs['standard_name'] = 'time'
+        dataarray['time'].attrs.pop('bounds', None)
+        dataarray = _add_time_dimension(dataarray)
+    return dataarray
+
+
+def _add_time_dimension(dataarray):
+    """Add 'time' dimension to the DataArray."""
+    # BUG: When number of timesteps equals the shape of y or x
```

Comment:
If this is a bug, please fix it. Do not comment about it.

Reply:
Fixed